### DISCUSSION TITLE
Kalman prediction rate row

### DISCUSSION THREAD

**Discussion by CalvinKoung** [bitcraze/discussions#1262] [Q&A]
Hi, I change the estimator_kalman.c to turn on the robustTwr. But I start to receive the Kalman Prediction rate low this warning. Why does this happen ?  I saw this blog (Robust Kalman Filter for Ultra-wideband Localization) and followed want to improved my crazyflie flight stability. Thanks
![VirtualBox_BitcrazeVM_02_04_2024_14_29_47](https://github.com/bitcraze/discussions/assets/118436188/71a5f33f-e032-438c-8798-f125b341139d)

**Comment by knmcguire** [comment]
Hi! The freertos task for the kalman state prediction is not running at the rate we initialized it to be, so there are perhaps some processes that take up soem resources. If you click 'task dump' button on the console you can perhaps find the curlprit. 

The prediction rate should be running at a 100 hz, but 98 is pretty okay too... so perhaps it is a bit of an strict warning
